### ISSUE TITLE
Get markdown file by file name. 

### ISSUE THREAD

**Issue by SachaG** [gatsbyjs/gatsby#9043]
I managed to query for a specific markdown file by its filename using this query:

```
    query($section: String) {
        allFile(
            filter: {
                internal: { mediaType: { eq: "text/markdown" } }
                name: { eq: $section }
            }
        ) {
            edges {
                node {
                    childMarkdownRemark {
                        html
                    }
                }
            }
        }
    }
```

(in this case `section` is the name of the file minus the extension). 

This works but it seems like a pretty complex query for something relatively basic. Is there a simpler way? Or is this how it should be done?

**Comment by stefanprobst** [comment]
The best way to explore graphql queries is on the GraphiQL console at http://localhost:8000/___graphql.

You can query for a file with
```graphql
query ($section: String) {
  file(name: {eq: $section}) {
    childMarkdownRemark {
      html
    }
  }
}
```
If necessary use additional filters like `relativePath` or `sourceInstanceName` or `extension`.

Alternatively, MarkdownRemark nodes have a `fileAbsolutePath` field you can filter by.

**Comment by SachaG** [comment]
Oh ok, I didn't know about `file` vs `allFile`. Makes sense. As for `MarkdownRemark` nodes, they do have `fileAbsolutePath` but no `name` as far as I can tell?

**Comment by stefanprobst** [comment]
> they do have `fileAbsolutePath` but no `name` as far as I can tell?

Yup, tbh i'm not really sure why `fileAbsolutePath` is there in the first place. I'd recommend doing filtering by name/dir on the file nodes.
